### PR TITLE
Homework4: Maksim Kvach

### PR DESCRIPTION
**Pull request by nkvamaks**
Seven tasks were created alongside with two text files:
test_cities.txt and test_languages.txt that contain input
data for corresponding tasks.

### PR checklist
Please make sure you've passed PR checklist
- [x] my PR passes CI checks (flake8 and pycodestyle)
- [x] my PR named according to guidelines.
- [x] my PR contains complete tasks descriptions for all tasks
- [ ] my PR made from the fork feature branch to maxim homework# branch.
- [x] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore

### CODE REVIEW

File: src/homework4/2_cities.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+relation = dict()  # stores relations: city-country(es)
+filename = 'test_cities.txt'  # input file
+
+with open(filename) as f:
+    # from input file creates a sort of database with a relation
+    # city-country(es)
+    for _N in range(int(f.readline())):
+        country, *cities = f.readline().split()
+        for city in cities:
+            relation[city] = relation.get(city, []) + [country]
```
Comment (belovmd) [inline src/homework4/2_cities.py:21]: append?

File: src/homework4/7_euclid_gcd.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+Даны два натуральных числа. Вычислите их наибольший общий делитель при помощи
+алгоритма Евклида (мы не знаем функции и рекурсию).
+"""
+# input two natural numbers and save their initial values
+first_int = int(input("Input 1st natural number: "))
+second_int = int(input("Input 2nd natural number: "))
+first_saved, second_saved = first_int, second_int
+
+if first_int < second_int:
```
Comment (belovmd) [inline src/homework4/7_euclid_gcd.py]: unclear naming

**Review by belovmd — COMMENTED** [review]
Thank you for your work. 
Please address next paragraphs from review notes [Review Notes](https://github.com/belovmd/it-academy-summer/blob/master/Homework%20review/Homework4.md):

- redundant comments in code
- 21

Overall it's the best work I've seen from your group.

File: src/homework4/2_cities.py (hunk 1 above)
Comment (nkvamaks) [reply src/homework4/2_cities.py:21]: Using 'append' was my first idea but it doesn't work in this situation.

File: src/homework4/7_euclid_gcd.py (hunk 2 above)
Comment (nkvamaks) [reply src/homework4/7_euclid_gcd.py]: Introduced clearer naming.

**Comment by nkvamaks** [comment]
Thank you for your review.
I've addressed your comments as much as I could: really redundant comments were deleted, but I've left some comments alongside with docstrings to clarify blocks of code. For you as a professional programmer these comments may still seem like redundant but for me as a beginner they remind what idea is inside the block of code, so please be gentle.
Task 4 was fixed.

**Comment by nkvamaks** [comment]
Just in case all 'redundant comments' were removed from code... Have no idea why you hate them so much...
Ngā mihi.

**Comment by belovmd** [comment]
> Just in case all 'redundant comments' were removed from code... Have no idea why you hate them so much...
> Ngā mihi.

If there are redundant comments in code - it's bad practice
If they are your temporary comments for yourself - it's just inaccurate

File: src/homework4/2_cities.py (hunk 1 above)
Comment (belovmd) [reply src/homework4/2_cities.py:21]: it should. this code looks weird

File: src/homework4/4_num_lists.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+один из этих списков.
+"""
+
+
+list1 = [1, 2, 3, 5, 99, 150, 151]
+list2 = [2, 3, 4, 6, 99, 111, 112]
+
+set1 = set(list1)
+set2 = set(list2)
+print(len(set1.symmetric_difference(set2)))
```
Comment (belovmd) [inline src/homework4/4_num_lists.py:13]: also ^

**Review by belovmd — COMMENTED** [review]
9,5

**Pull request closed without merging**